main.go

```diff
@@ -0,0 +1,196 @@
+		panic(err)
+	}
+
+	fmt.Println("Connected to MongoDB!")
+
+	// Provide the name of the database and collection you want to use.
+	// If they don't already exist, the driver and Atlas will create them
+	// automatically when you first write data.
+	var dbName = "myDatabase"
+	var collectionName = "recipes"
+	collection := client.Database(dbName).Collection(collectionName)
+
+	/*      *** INSERT DOCUMENTS ***
+	 *
+	 * You can insert individual documents using collection.Insert().
+	 * In this example, we're going to create 4 documents and then
+	 * insert them all in one call with InsertMany().
+	 */
+
+	eloteRecipe := Recipe{
+		Name:              "elote",
+		Ingredients:       []string{"corn", "mayonnaise", "cotija cheese", "sour cream", "lime"},
+		PrepTimeInMinutes: 35,
+	}
+
+	locoMocoRecipe := Recipe{
+		Name:              "loco moco",
+		Ingredients:       []string{"ground beef", "butter", "onion", "egg", "bread bun", "mushrooms"},
+		PrepTimeInMinutes: 54,
+	}
+
+	patatasBravasRecipe := Recipe{
+		Name:              "patatas bravas",
+		Ingredients:       []string{"potato", "tomato", "olive oil", "onion", "garlic", "paprika"},
+		PrepTimeInMinutes: 80,
+	}
+
+	friedRiceRecipe := Recipe{
+		Name:              "fried rice",
+		Ingredients:       []string{"rice", "soy sauce", "egg", "onion", "pea", "carrot", "sesame oil"},
+		PrepTimeInMinutes: 40,
+	}
+
+	// Create an interface of all the created recipes
+	recipes := []interface{}{eloteRecipe, locoMocoRecipe, patatasBravasRecipe, friedRiceRecipe}
+	insertManyResult, err := collection.InsertMany(context.TODO(), recipes)
+	if err != nil {
+		fmt.Println("Something went wrong trying to insert the new documents:")
+		panic(err)
+	}
+
+	fmt.Println(len(insertManyResult.InsertedIDs), "documents successfully inserted.")
+
+	/*
+	 * *** FIND DOCUMENTS ***
+	 *
+	 * Now that we have data in Atlas, we can read it. To retrieve all of
+	 * the data in a collection, we create a filter for recipes that take
+	 * less than 45 minutes to prepare and sort by name (ascending)
+	 */
+
+	var filter = bson.M{"prepTimeInMinutes": bson.M{"$lt": 45}}
+	options := options.Find()
+
+	// Sort by `name` field ascending
+	options.SetSort(bson.D{{"name", 1}})
+
+	cursor, err := collection.Find(context.TODO(), filter, options)
+	if err != nil {
+		fmt.Println("Something went wrong trying to find the documents:")
+		panic(err)
+	}
+
+	defer func() {
+		cursor.Close(context.Background())
+	}()
+
+	// Loop through the returned recipes
+	for cursor.Next(ctx) {
+		recipe := Recipe{}
+		err := cursor.Decode(&recipe)
+
+		// If there is an error decoding the cursor into a Recipe
+		if err != nil {
+			fmt.Println("cursor.Next() error:")
+			panic(err)
+		} else {
+			fmt.Println(recipe.Name, "has", len(recipe.Ingredients), "ingredients, and takes", recipe.PrepTimeInMinutes, "minutes to make.")
+		}
+	}
+
+	// We can also find a single document. Let's find the first document
+	// that has the string "fried rice" in the name.
```

Comment:
Nitpick: you'll want to update this message to match the new filter.

Reply:
👍